src/usecases/retrieveWardVisitTotals.js

```diff
@@ -1,35 +1,41 @@
 const calculateTotalNumberOfVisits = (visitsByWard) =>
   visitsByWard.reduce((total, { visits }) => total + visits, 0);
 
-const getVisitsByWard = async (db) => {
+const getVisitsByWard = async (db, trustId) => {
   const queryResult = await db.any(
     `SELECT
-      wards.name,
-      SUM(totals.total) AS total_visits,
-      (
-        SELECT
-          name
-        FROM
-          hospitals
-        WHERE
-          id = wards.hospital_id
-      ) as hospital_name,
-      wards.hospital_id as hospital_id
-    FROM ward_visit_totals AS totals JOIN wards ON wards.id = totals.ward_id
-    GROUP BY wards.hospital_id, wards.name`
+    wards.name,
+    SUM(totals.total) AS total_visits,
+    (
+      SELECT
+        name
+      FROM
+        hospitals
+      WHERE
+        id = wards.hospital_id
+    ) as hospital_name,
+    wards.hospital_id as hospital_id,
+    wards.trust_id as trust_id
+    FROM ward_visit_totals AS totals JOIN wards ON wards.id = totals.ward_id AND ($1 IS NULL OR wards.trust_id = $1)
+    GROUP BY wards.trust_id, wards.hospital_id, wards.name
```

Comment:
Should we be grouping by `wards.name` here or `wards.id` ?  
Although likely rare, could we end up grouping two different wards with the same name (since it's not a unique field) ?

Reply:
Nice spot! 🕵️‍♂️  Will update.